nmea: add GPS sentence code to nmea package

trekhopton opened the pull request:
This code was previously in a client program, now it can be used in this package by other repos.

trekhopton committed:
nmea: add GPS sentence code to nmea

This code was previously in a client program, now it can be used in this package by other repos.

ao-david commented on nmea/gps.go:
```diff
@@ -0,0 +1,114 @@
+	gonmea "github.com/adrianmo/go-nmea"
+)
+
+// GPSData stores data collected from NMEA sentences
```
fullstop :)

ao-david commented on nmea/gps.go:
```diff
@@ -0,0 +1,114 @@
+	True             *bool      // Heading is relative to true north
+}
+
+func ProcessSentence(dst GPSData, raw string) (GPSData, error) {
```
Exported function needs comment

ao-david commented on nmea/gps.go:
```diff
@@ -0,0 +1,114 @@
+	case gonmea.RMC:
+		if s.Validity != "A" {
+			// Not valid data.
+			return dst, errors.New("invalid data in RMC sentence")
```
Make this error an exported constant

trekhopton replied:
Why do you suggest this? Not saying it's a bad idea.

ao-david replied:
Means the caller can check the error and may implement different flows.

ao-david commented on nmea/gps.go:
```diff
@@ -0,0 +1,114 @@
+	case gonmea.GLL:
+		if s.Validity != "A" {
+			// Not valid data.
+			return dst, errors.New("invalid data in GLL sentence")
```
ditto

ao-david commented on nmea/gps.go:
```diff
@@ -0,0 +1,114 @@
+package nmea
```
Add package header (copyright etc...)

ao-david reviewed (approved):
I assume that most of the logic has just been copied across, I'm not familiar with NMEA to validate.

Is there a reason why you both pass in a pointer to the dst and then also return the pointer? is this just because thats how the package that this was copied from did it?

Just a few comments and stuff to cleanup

trekhopton commented:
See https://github.com/ausocean/client/pull/88 .

trekhopton committed:
nmea: simplify GGA sentence parsing code.

This was done to make it more readable and maintainable. It specifies it so that it is more relevant to what we want to do.
